pil-analyzer/tests/condenser.rs

```diff
@@ -174,3 +174,95 @@ pub fn constructed_constraints() {
 "#;
     assert_eq!(formatted, expected);
 }
+
+#[test]
+#[should_panic = "There are pre-existing constraints or witness columns"]
+pub fn capture_stage_non_fresh_wit() {
+    let input = r#"
+        namespace std::prover;
+            let capture_stage: (-> int) -> Constr[] = 9;
+
+        namespace Main(1024);
+            col witness x;
+            let f = constr || { x = 2; 1024 };
+            std::prover::capture_stage(f);
+    "#;
+    analyze_string::<GoldilocksField>(input);
+}
```

Comment:
The idea is that you call `capture_stage` to capture a full stage. This means that before the call to capture_stage, there cannot be any constraints or witness columns in the environment. Since capture_stage increments the stage, they would have a different stage index than those after the call to capture_stage.

I would also prefer to do this differently so that we don't need this and assign the correct stage automatically, but I don't know how to do this currently, and I also don't think that this would occur often.